src/msg/async/dpdk/dpdk_rte.cc

```diff
@@ -45,6 +45,29 @@ namespace dpdk {
     return std::bitset<CHAR_BIT * sizeof(n)>{n}.count();
   }
 
+  uint16_t eal::get_first_slave_coreid(const char *coremask)
+  {
+    uint64_t ulCoreMask;
+    uint16_t usBitCount;
+    uint16_t first_slave_coreid;
+    int iRet = 0;
+
+    iRet = sscanf(coremask, "%lx", &ulCoreMask);
+    if (1 != iRet || 0 == ulCoreMask)
+    {
+        return 0;
+    }
+
+    for (usBitCount = 0; 0 == ulCoreMask % 2 && 0 != ulCoreMask; usBitCount++)
```

Comment:
no need to reinvent the wheel, could just use the templatized  `clz<>()` for counting the leading zeros.